sdks/java/core/src/main/java/org/apache/beam/sdk/io/common/retry/BaseRetryConfiguration.java

```diff
@@ -0,0 +1,52 @@
+/*
```

Comment:
@echauchot 
Moving withRetryPredicate to the BaseRetryConfiguration will expose it in the API. Currently it is package private, visibleForTesting which was by design in SolrIO.

Reply:
@timrobertson100 @echauchot i agree moving withRetryPredicate to BaseRetryConfiguration will expose API, the only way i can avoid is by overriding the method with protected final key word in ESIO or SolrIO. This will lead to duplicate code in all the implementations which use the common configuration classes.

I think it is better to have duplicates with AutoValue Builder in ESIO and SolrIO instead of using common config objects. I will remove common code.

Reply:
@timrobertson100 you're right about exposing withRetryPredicate, it is better to hide it from the user and keep this knowledge at the IO level.